### PR TITLE
Bug-Fix: Patient's Age parsing discrepancy during XLS export in Sample Management System

### LINKED ISSUE #9231: Bug: Exported XLS files has mismatched Patient's Age (Sample Test Management)
**Describe the bug**
While exploring the features I came across this bug. Like when we export the Samples Data into excel sheet, everything seems fine, but some of the `Patient's Age` are not calculated in properly and hence parsed in with some random date in the excel cells.

**To Reproduce**

1. Go to 'Sample Management System'
2. Click on 'Export Button'
3. Open the XLS sheet
4. Have a look on the `Patient's Age` column.
5. Some Ages are not parsed properly. 

**Expected behavior**
The Patient Ages should be properly populated accordingly.

**Screenshots**
Image

**Desktop (please complete the following information):**

- OS: Linux
- Browser: Chrome, Edge

**Proposed Solution**

- For some records (could be some XLS parsing issue) the Patient's Age is interpreted as `Date` but not as the `string`.
- We need to handle them manually such that if they are not parsed as `Date` but as strings.

https://github.com/ohcnetwork/care_fe/blob/fc2ac3b0f66775c05e36a012823125228271fdfd/src/components/Patient/SampleViewAdmin.tsx#L125-L131

Have a look onto this and lemme know if I can work on fixing this issue. @Jacobjeevan 

cc @nihal467 @rithviknishad

### PR DESCRIPTION
**Pull request by pooranjoyb**
## Proposed Changes

- Fixes #9231 
- Manually parsed the data to resolve the error.
- Conditions are made based on column name and logic written accordingly.

@ohcnetwork/care-fe-code-reviewers

### Screenshots of the fix : 

image

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

@Jacobjeevan Please have a look if this is good to go. Thank You :) 

## Summary by CodeRabbit

- **New Features**
	- Improved data export functionality in the Sample View Admin component for better formatting and readability of patient data.

- **Bug Fixes**
	- Enhanced processing of date fields for consistency and accuracy, including better error handling for invalid dates.

### CODE REVIEW

**Review by Jacobjeevan — APPROVED** [review]
LGTM

**Comment by pooranjoyb** [comment]
Hii @nihal467 please check if this is good to go. :)

**Review by rithviknishad — APPROVED** [review]
LGTM, but I'd have recommended building a reusable utility function that could do something like shown, but this is also fine for now.

```
transformCSVData(
  data,
  {
    dateFields: ['field1', 'field2'],
    customTransforms: [
      ['field1']: (cell) => output
    ]
  }
)
```

**Comment by nihal467** [comment]
image

@pooranjoyb when the date of result is empty, its been shown as invalid date, if the data is not available, leave the column blank

**Comment by pooranjoyb** [comment]
Okaay Updating it asap. @nihal467

**Comment by pooranjoyb** [comment]
> @pooranjoyb when the date of result is empty, its been shown as invalid date, if the data is not available, leave the column blank

Changes are made as requested. Please have a look sir @nihal467 

image

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
image
image

@pooranjoyb when the patient age <1, the patient age should be shown as month and days , as we show in the patient list page,

**Comment by pooranjoyb** [comment]
> @pooranjoyb when the patient age <1, the patient age should be shown as month and days , as we show in the patient list page, 

Okaayy, will update this asap. Thanks for the review @nihal467

**Comment by pooranjoyb** [comment]
Hey @nihal467 (sorry for the delay)
But it seems the age of the Patient is being directly fetched from the BE and not being handled here in FE.

This is the 3rd record in `devdoctor` account.
image

You can see the data is fetched directly from BE.
**https://github.com/ohcnetwork/care_fe/blob/f87a2a4d1cd330b5ae1a2d8f2480a224abf26568/src/components/Patient/SampleViewAdmin.tsx#L323-L332**


Please lemme know if I'm missing something here. @nihal467 @rithviknishad @Jacobjeevan

**Comment by nihal467** [comment]
@rithviknishad can you look into the issue

**Comment by rithviknishad** [comment]
@nihal467 We can show 0 itself right? Changes in the back-end will be required for this since we are not getting the DOB of the patient in this API.

File: src/components/Patient/SampleViewAdmin.tsx [hunk 1]
```diff
@@ -117,22 +117,31 @@ export default function SampleViewAdmin() {
...
-          .map((field: string) =>
-            new Date(field).toString() === "Invalid Date"
-              ? field
-              : formatDateTime(field),
-          )
-          .join(","),
-      )
-      .join("\n");
+  const parseExportData = (data: string) => {
+    const rows = data.trim().split("\n");
```
Comment (rithviknishad) [inline src/components/Patient/SampleViewAdmin.tsx]: btw cleaner approach:

```suggestion
    const [header, ...rows] = data.trim().split("\n");
```

**Comment by pooranjoyb** [comment]
> @nihal467 We can show 0 itself right? Changes in the back-end will be required for this since we are not getting the DOB of the patient in this API.

Yes, of course 0 can be done. If you'd ask, I can go with the requested changes.

File: src/components/Patient/SampleViewAdmin.tsx (hunk 1 above)
Comment (pooranjoyb) [reply src/components/Patient/SampleViewAdmin.tsx]: This requested change is done.

**Comment by pooranjoyb** [comment]
@rithviknishad @nihal467 Requested changes are pushed. Please review. Thanks :)

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by Jacobjeevan — APPROVED** [review]

**Pull request merged**